protostar/testing/cheatcodes/send_message_to_l2.py

```diff
@@ -0,0 +1,68 @@
+from typing import Callable, Any, Mapping, Optional
+
+from starkware.starknet.business_logic.execution.execute_entry_point import (
+    ExecuteEntryPoint,
+)
+from starkware.starknet.business_logic.execution.objects import CallType
+from starkware.starknet.public.abi import get_selector_from_name
+from starkware.starknet.services.api.contract_class import EntryPointType
+
+from protostar.starknet import Cheatcode, CheatcodeException
+from protostar.utils.data_transformer import CairoOrPythonData, from_python_transformer
+
+
+class SendMessageToL2Cheatcode(Cheatcode):
+    @property
+    def name(self) -> str:
+        return "send_message_to_l2"
+
+    def build(self) -> Callable[..., Any]:
+        return self.send_message_to_l2
+
+    def send_message_to_l2(
+        self,
+        fn_name: str,
+        l1_sender_address: int = 0,
+        contract_address: Optional[int] = None,
+        calldata: Optional[CairoOrPythonData] = None,
+    ) -> None:
+        contract_address = (
+            contract_address if contract_address else self.contract_address
+        )
+
+        class_hash = self.state.get_class_hash_at(contract_address)
+        contract_class = self.state.get_contract_class(class_hash)
+
+        assert contract_class.abi, "Contract ABI not available"
```

Comment:
[Nit, Functionality] I think this should be a `CheatcodeException`. Contract ABI is empty, if the contract doesn't have any entrypoints (@external, @view, etc.). A user-friendly error description would be more helpful.

Reply:
Good point. Will implement